languages/go/tree-sitter-go.cabal

```diff
@@ -13,6 +13,15 @@ cabal-version:       >=1.10
 
 library
   exposed-modules:     TreeSitter.Go
+  build-depends:       base >= 4.7 && < 5
+                     , haskell-tree-sitter
+                     , template-haskell
+                     , tree-sitter-go-internal
+  default-language:    Haskell2010
+
+library tree-sitter-go-internal
+  exposed-modules:     TreeSitter.Go.Internal
+  hs-source-dirs:      internal
```

Comment:
We additionally have to define an internal library which builds the C sources so that the public library can depend on that, and therefore be assured that the required `_tree_sitter_*` symbol is available when the TH code is run.